lib/models/referral.js

```diff
@@ -139,20 +145,34 @@ Referral.statics.create = function(marketing, recipientEmail, type) {
       return reject(new errors.BadRequestError('Invalid marketing doc'));
     }
 
-    var referral = new Referral({
-      sender: {
-        id: marketing._id,
-        referralLink: marketing.referralLink
-      },
-      recipient: {
-        email: recipientEmail
-      },
-      type
-    });
+    Referral.findOne({
+      'sender.id': marketing._id,
+      'recipient.email': recipientEmail
+    }).then((referral) => {
+      if (referral) {
+        referral.count += referral.count;
```

Comment:
There may be a concurrency issue here if two processes do this at the same time, one of the count will go missing.